iOS/Flows/Conversation/_New/ConversationViewController.swift

```diff
@@ -172,83 +223,83 @@ class ConversationViewController: FullScreenViewController,
+            self.collectionView.alpha = 0.5
         }
 
-        let messages = controller.messages
-        var snapshot = self.dataSource.snapshot()
+        switch position {
+        case .left, .middle:
+            // Avoid animating content offset twice for redundant states
+            guard self.lastPreparedPosition.equalsOneOf(these: .left, .middle) else { break }
 
-        snapshot.appendSections([.loadMore])
-        if !controller.hasLoadedAllPreviousMessages {
-            snapshot.appendItems([.loadMore], toSection: .loadMore)
+            if let initialContentOffset = self.initialContentOffset {
+                self.collectionView.setContentOffset(initialContentOffset, animated: true)
+            }
+        case .right:
+            let newXOffset = self.collectionView.contentSize.width - self.layout.minimumLineSpacing
+            self.collectionView.setContentOffset(CGPoint(x: newXOffset, y: 0), animated: true)
         }
 
-        snapshot.appendSections([.conversation(conversation.cid)])
-        snapshot.appendItems(messages.asConversationCollectionItems)
-        
-        let animationCycle = AnimationCycle(inFromPosition: .left,
-                                            outToPosition: .right,
-                                            shouldConcatenate: true,
-                                            scrollToEnd: true)
-        
-        await self.dataSource.apply(snapshot,
-                                    collectionView: self.collectionView,
-                                    animationCycle: animationCycle)
+        self.lastPreparedPosition = position
     }
-    
-    func subscribeToConversationUpdates() {
-        self.conversationController.messagesChangesPublisher.mainSink { [unowned self] changes in
-            Task {
-                guard let conversationController = self.conversationController else { return }
-                await self.dataSource.update(with: changes,
-                                             conversationController: conversationController,
-                                             collectionView: self.collectionView)
-            }
-        }.store(in: &self.cancellables)
 
-        self.conversationController.channelChangePublisher.mainSink { [unowned self] _ in
-            self.conversationHeader.configure(with: self.conversation)
-        }.store(in: &self.cancellables)
+    func swipeableInputAccessoryDidUnprepareSendable(_ view: SwipeableInputAccessoryView) {
+        UIView.animate(withDuration: Theme.animationDuration) {
+            self.collectionView.alpha = 1
+        }
 
-        self.conversationController.memberEventPublisher.mainSink { [unowned self] _ in
-            self.conversationHeader.configure(with: self.conversation)
-        }.store(in: &self.cancellables)
+        guard let initialContentOffset = self.initialContentOffset else { return }
+        self.collectionView.setContentOffset(initialContentOffset, animated: true)
```

Comment:
"Unprepare" means they dragged back down, in that case set the content offset back to where we started.